$OpenBSD: patch-parser-specs_config_spec,v 1.3 2013/01/27 18:33:20 dcoppa Exp $

Workaround for an error caused by xcb-util-0.3.6
In file included from src/config_parser.c:49:
include/GENERATED_config_enums.h:70: error: 'FONT' redeclared as different kind of symbol
/usr/X11R6/include/xcb/xcb_atom.h:60: error: previous declaration of 'FONT' was here

Bugfix: parse outputs as "word", not "string", to ignore trailing
whitespaces (upstream git commit 8bc771ac16362f05d98bf8b29877939d6edb3c91)

Fix bind[code|sym] --release
(upstream git commit 6dfb08170cb1b0bd1875b80c85fbc225ff22274d)

--- parser-specs/config.spec.orig	Wed Dec 12 00:08:17 2012
+++ parser-specs/config.spec	Sun Jan 27 19:04:57 2013
@@ -20,7 +20,7 @@ state INITIAL:
   'set'                                    -> IGNORE_LINE
   bindtype = 'bindsym', 'bindcode', 'bind' -> BINDING
   'bar'                                    -> BARBRACE
-  'font'                                   -> FONT
+  'font'                                   -> WINDOW_FONT
   'mode'                                   -> MODENAME
   'floating_minimum_size'                  -> FLOATING_MINIMUM_SIZE_WIDTH
   'floating_maximum_size'                  -> FLOATING_MAXIMUM_SIZE_WIDTH
@@ -214,7 +214,7 @@ state WORKSPACE_OUTPUT:
       -> WORKSPACE_OUTPUT_STR
 
 state WORKSPACE_OUTPUT_STR:
-  output = string
+  output = word
       -> call cfg_workspace($workspace, $output)
 
 # ipc-socket <path>
@@ -264,12 +264,14 @@ state EXEC:
       -> call cfg_exec($exectype, $no_startup_id, $command)
 
 # font font
-state FONT:
+state WINDOW_FONT:
   font = string
       -> call cfg_font($font)
 
 # bindsym/bindcode
 state BINDING:
+  release = '--release'
+      ->
   modifiers = 'Mod1', 'Mod2', 'Mod3', 'Mod4', 'Mod5', 'Shift', 'Control', 'Ctrl', 'Mode_switch'
       ->
   '+'
@@ -313,6 +315,8 @@ state MODE_IGNORE_LINE:
       -> MODE
 
 state MODE_BINDING:
+  release = '--release'
+      ->
   modifiers = 'Mod1', 'Mod2', 'Mod3', 'Mod4', 'Mod5', 'Shift', 'Control', 'Ctrl', 'Mode_switch'
       ->
   '+'
